### PR TITLE
test(cb2-5882): add in effect tests

### PR DESCRIPTION
**Pull request by naathanbrown**
##  add in effect tests

Add in missed effect tests
## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts [hunk 1]
```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
...
+        jest.spyOn(technicalRecordService, 'postProvisionalTechRecord').mockReturnValue(cold('--a|', { a: technicalRecord[0] }));
+
+        // expect effect to return success action
+        expectObservable(effects.postProvisionalTechRecord).toBe('---b', {
+          b: createProvisionalTechRecordSuccess({ vehicleTechRecords: technicalRecord })
+        });
+      });
+    });
+
+    it('should return an error message if not not found', () => {
```
Comment (aaron-downing) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:575]: Is there a reason the error codes are different between the two calls? 500 for update and 400 for post provisional?

Comment (naathanbrown) [reply src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:575]: Yes, I didn't push save when testing something, thank you!

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts [hunk 2]
```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
     });
   });
 
-  // TODO: Unit tests for putUpdate and postProvisional methods
+  describe('updateTechnicalRecord$', () => {
+    it('should return a technical record on successfull API call', () => {
```
Comment (tahayazici) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:519]: So sorry, but it is one of those...
```suggestion
    it('should return a technical record on successful API call', () => {
```

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts [hunk 3]
```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
...
+        jest.spyOn(technicalRecordService, 'putUpdateTechRecords').mockReturnValue(cold('--a|', { a: technicalRecord[0] }));
+
+        // expect effect to return success action
+        expectObservable(effects.updateTechnicalRecord$).toBe('---b', {
+          b: updateTechRecordsSuccess({ vehicleTechRecords: technicalRecord })
+        });
+      });
+    });
+
+    it('should return an error message if not not found', () => {
```
Comment (tahayazici) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:536]:
```suggestion
    it('should return an error message if not found', () => {
```

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts [hunk 4]
```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
...
+      testScheduler.run(({ hot, cold, expectObservable }) => {
+        // mock action to trigger effect
+        actions$ = hot('-a--', { a: updateTechRecords });
+
+        // mock service call
+        const expectedError = new HttpErrorResponse({
+          status: 500,
+          statusText: 'Internal server error'
+        });
```
Comment (tahayazici) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:545]: Optional, but this only has 2 simple properties. You could chuck it in one line.
```suggestion
        const expectedError = new HttpErrorResponse({ status: 500, statusText: 'Internal server error' });
```

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts [hunk 5]
```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
...
+          b: updateTechRecordsFailure({
+            error: 'Unable to update technical record null'
+          })
+        });
+      });
+    });
+  });
+
+  describe('postProvisionalTechRecord$', () => {
+    it('should return a technical record on successfull API call', () => {
```
Comment (tahayazici) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:558]:
```suggestion
    it('should return a technical record on successful API call', () => {
```

File: src/app/store/technical-records/effects/technical-record-service.effects.spec.ts (hunk 1 above)
Comment (tahayazici) [inline src/app/store/technical-records/effects/technical-record-service.effects.spec.ts:575]:
```suggestion
    it('should return an error message if not found', () => {
```

**Review by tahayazici — CHANGES_REQUESTED** [review]
Small comments to address some typos. Good work!

**Review by kpearmanbjss — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Review by tahayazici — APPROVED** [review]

**Pull request merged**